

```
Name: Satya Sudha
Week 11
Xgboost with 5 CV
```



In [1]:
# Importing the required packages
import pandas as pd
import numpy as np
import time
from sklearn.model_selection import cross_val_score
from xgboost import XGBClassifier


In [3]:
#Loading the datasets
data_100 = pd.read_csv('data_100.csv')
data_1000 = pd.read_csv('data_1000.csv')
data_10000 = pd.read_csv('data_10000.csv')
data_100000 = pd.read_csv('data_100000.csv')
data_1000000 = pd.read_csv('data_1000000.csv')
data_10000000 = pd.read_csv('data_10000000.csv')

In [4]:
datasets = {
    100: data_100,
    1000: data_1000,
    10000: data_10000,
    100000: data_100000,
    1000000: data_1000000,
    10000000: data_10000000
}

results = []

for size, df in datasets.items():
    print(f"▶ Working on dataset of size {size}")

    X = df.drop(columns='outcome')
    y = df['outcome']

    model = XGBClassifier(
        use_label_encoder=False,
        eval_metric='logloss',
        verbosity=0
    )

    start_time = time.time()

    scores = cross_val_score(model, X, y, cv=5, scoring='accuracy')
    mean_accuracy = np.mean(scores)

    end_time = time.time()
    elapsed_time = end_time - start_time

    results.append({
        'Method used': 'XGBoost (Python, sklearn, 5-fold CV)',
        'Dataset size': size,
        'Testing-set predictive performance': mean_accuracy,
        'Time taken for the model to be fit (seconds)': elapsed_time
    })

# Converting results to DataFrame
results_df = pd.DataFrame(results)

# Prining the results
print(results_df)

# Saving in the table format
results_df.to_csv('xgboost_python_results.csv', index=False)


▶ Working on dataset of size 100
▶ Working on dataset of size 1000
▶ Working on dataset of size 10000
▶ Working on dataset of size 100000
▶ Working on dataset of size 1000000
▶ Working on dataset of size 10000000
                            Method used  Dataset size  \
0  XGBoost (Python, sklearn, 5-fold CV)           100   
1  XGBoost (Python, sklearn, 5-fold CV)          1000   
2  XGBoost (Python, sklearn, 5-fold CV)         10000   
3  XGBoost (Python, sklearn, 5-fold CV)        100000   
4  XGBoost (Python, sklearn, 5-fold CV)       1000000   
5  XGBoost (Python, sklearn, 5-fold CV)      10000000   

   Testing-set predictive performance  \
0                            0.860000   
1                            0.942000   
2                            0.975300   
3                            0.987300   
4                            0.991825   
5                            0.993127   

   Time taken for the model to be fit (seconds)  
0                                      0.387820  

As the dataset size increases, the XGBoost model's performance improves significantly. On smaller datasets, such as the one with just 100 samples, the model achieved an accuracy of 86%. However, as the dataset grows, so does the accuracy, reaching 99.31% on the largest dataset of 10 million samples. This demonstrates that the model becomes more accurate with more data, which is expected for machine learning models. Along with the improved accuracy, the time taken to fit the model also increases with the size of the dataset. For the 100-sample dataset, the model was trained in just 0.39 seconds, while for the 10 million-sample dataset, it took approximately 454 seconds (or 7.5 minutes). This trade-off between training time and performance is important to consider: smaller datasets allow for faster training and good accuracy, but larger datasets provide even better accuracy at the cost of longer computation times.